External/FEXCore/Source/Interface/Core/Interpreter/ALUOps.cpp

```diff
@@ -493,6 +493,28 @@ DEF_OP(Extr) {
   }
 }
 
+DEF_OP(PExt) {
+  const auto Op = IROp->C<IR::IROp_PExt>();
+  const auto OpSize = IROp->Size;
+
+  if (OpSize != 4 && OpSize != 8) {
+    LOGMAN_MSG_A_FMT("Unknown PExt Size: {}\n", OpSize);
+    return;
+  }
+
+  const auto Input = *GetSrc<uint64_t*>(Data->SSAData, Op->Args(0));
+  auto Mask = *GetSrc<uint64_t*>(Data->SSAData, Op->Args(1));
+
+  uint64_t Result = 0;
+  for (uint64_t Offset = 0; Mask > 0; Offset++) {
```

Comment:
Does this correctly handle the case when the mask is set to a 64-bit mask, then operates on the 32-bit sized registers? Little bit pedantic but I want to make sure.
eg:
```
mov rax, 0x1234567812345678
mov rbi, 0xFF00FF00FF00FF00
pext ecx, eax, ebx
```